packages/core/src/account/smartContractAccount.ts

```diff
@@ -164,12 +175,13 @@ export const getAccountAddress = async ({
 export async function toSmartContractAccount<
   Name extends string = string,
   TTransport extends Transport = Transport,
-  TChain extends Chain = Chain
+  TChain extends Chain = Chain,
+  TEntryPointVersion extends EntryPointVersion = EntryPointVersion
 >({
   transport,
   chain,
   source,
-  entryPoint = getVersion060EntryPoint(chain),
+  entryPoint = getEntryPoint(chain),
```

Comment:
the current definition of `getEntryPoint` isn't typesafe because the `TEntryPointVersion` doesn't necessarily line up with the version being returned by this. someone could specify `0.6.0` for `TEntryPointVersion` but there's nothing stopping someone from calling this with `getEntryPoint(chain, "0.7.0")`

Reply:
oh never mind just saw the updated params